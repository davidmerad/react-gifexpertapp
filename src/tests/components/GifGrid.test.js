import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en <GifGrid />', () => {
    let category = 'One Punch';
    // let wrapper = shallow(<GifGrid category={category} />);

    // beforeEach(()=>{
    //     wrapper =  shallow(<GifGrid category={category} />);
    // });

    test('Debe mostrar correctamente el <GifGrid />', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [
            {
                id: 'ABC',
                title: 'cualquier cosa',
                url: 'https://log.com/img.jpg'
            },
            {
                id: 'ABCD',
                title: 'cualquier cosa',
                url: 'https://log.com/img.jpg'
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
    
})
