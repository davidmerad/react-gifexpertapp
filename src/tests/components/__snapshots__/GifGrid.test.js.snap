// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar correctamente el <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      loading...
    </p>
  </div>
</Fragment>
`;
